packages/webdriverio/tests/commands/browser/url.test.js

```diff
@@ -27,14 +27,39 @@ describe('url', () => {
             .toEqual({ url: 'http://foobar.com/foobar' })
     })
 
-    it('should throw an exception when a non-string value passed in', async () => {
+    it('should throw an exception when a non-string value passed in path', async () => {
         try {
             await browser.url(true)
         } catch (e) {
             expect(e.message).toContain('command needs to be type of string')
         }
     })
 
+    it('should throw an exception when a non-function value passed in inject', async () => {
+        try {
+            await browser.url('/', { inject: 123 })
+        } catch (e) {
+            expect(e.message).toContain('command needs to be type of function')
+        }
+    })
+
+    it('should continue the navigation if inject parameter is not passed', async () => {
+        await browser.url('/')
+        expect(got.mock.calls[0][1].json).toEqual({ url: 'http://foobar.com/' })
+    })
+
+    it('should accept an arrow function as the inject parameter', async () => {
+        const arrowFunction = jest.fn(() => console.log('foo-bar'))
+        await browser.url('/', { inject: arrowFunction })
+        expect(got.mock.calls[0][1].json).toEqual({ url: 'http://foobar.com/' })
+    })
+
+    it('should accept a normal function as the inject parameter', async () => {
+        const arrowFunction = jest.fn(function () { console.log('foo-bar') })
+        await browser.url('/', arrowFunction)
```

Comment:
Since the arrowFunction isn't used at all for any assertions, simply do the following instead:
```suggestion
        await browser.url('/', () => {})
```

Reply:
@erwinheitzman I got that. However, if you see @christian-bromann comments, I am lagging on how to verify whether puppeteer was used to inject the function.

Is there a way, where I can verify puppeteer instance is created? The above test will create the puppeteer instance and I can add a step in the above function inject tests to check instance is created or not.

Reply:
It will not use the webdriver protocol so I would say that you can assert the mock calls

Reply:
I think it is useful to pass in a parameter so that the assertion is easier:

```js
const urlParams = { inject: jest.fn() }
await browser.url('/', urlParams)
...
expect(pageMock.evaluate).toBeCalledWith(urlParams.inject)
```